src/protocol_traits/graph.rs

```diff
@@ -136,6 +136,12 @@ pub trait Graph: Default {
     fn set_edge_metadata(&mut self, edge_id: &Self::EdgeId, new: Self::EdgeMetadata);
 }
 
+/// Trait to print (parts of) the for debugging purposes
+pub trait PrintableGraph: Graph {
```

Comment:
I would suggest moving this trait somewhere else, so that we make clear this is not an "official" protocol trait, but more something we added for own own debugging purposes. I think even `types.rs` would be a good home for now, we could think later if we need a more specialised module for these traits 😉

Reply:
:+1: Moving it into `types.rs`